### DISCUSSION TITLE
Nats jetstream publish async

### DISCUSSION THREAD

**Discussion by lisablack1** [nats-io/nats.go#1144] [Q&A] [answered]
Hi can some one provide any better example for nats publish in asynchronous

i have tried in the below format , its keep on going to else condition , 

streamContext, err := connection.JetStream(nats.PublishAsyncMaxPending(1000)) <<--- wait for 1000 ack pending msgs 

am i doing anything wrong , any suggestions would be appreciated 

        for i:=0;i<100000;i++{
            if streamContext.PublishAsyncPending() < 1000 {
                _, err = streamContext.PublishAsync(newSubjectName, bytesInfo) //sample msg in bytes format
        
                if err != nil {
                    log.Println("publish error:", err)
                    continue
                }
                log.Println("msg published")
            } else {
                log.Println("PublishAsyncPending:", streamContext.PublishAsyncPending())
                time.Sleep(time.Millisecond * 10)
                continue
            }
        }

**Comment by derekcollison** [comment, marked as answer]
The system will automatically gate if the pubacks outstanding is >= the max, no need to test.

<img width="468" alt="image" src="https://user-images.githubusercontent.com/90097/203823008-218ecd6b-8d96-41bf-8b60-d4edf98b0814.png">

**Comment by lisablack1** [comment]
what is maximum value for AsyncMaxPending we can provide ? , will it cause any blocks or holds any memory if it reaches to the max  and waiting for acks

**Comment by derekcollison** [comment]
No limits, will just hold all pending acks in memory so they can be checked. If the client reaches the max it will wait til pending is below the threshold.
